zeek/bin/spicyz.cc

```diff
@@ -0,0 +1,226 @@
+                break;
+
+            case 'C': {
+                driver_options->dump_code = true;
+                break;
+            }
+
+            case 'd': {
+                compiler_options->debug = true;
+                break;
+            }
+
+            case 'X': {
+                auto arg = std::string(optarg);
+
+                if ( arg == "help" ) {
+                    std::cerr << "Additional debug instrumentation:\n";
+                    std::cerr << "   flow:     log function calls to debug stream \"hilti-flow\"\n";
+                    std::cerr << "   location: track current source code location for error reporting\n";
+                    std::cerr << "   trace:    log statements to debug stream \"hilti-trace\"\n";
+                    std::cerr << "\n";
+                    exit(0);
+                }
+
+                compiler_options->debug = true;
+
+                if ( auto r = compiler_options->parseDebugAddl(arg); ! r )
+                    return hilti::result::Error(r.error());
+
+                break;
+            }
+
+            case 'D': {
+                auto arg = std::string(optarg);
+
+                if ( arg == "help" ) {
+                    std::cerr << "Debug streams:\n";
+
+                    for ( const auto& s : hilti::logging::DebugStream::all() )
+                        std::cerr << "  " << s << "\n";
+
+                    std::cerr << "\n";
+                    exit(0);
+                }
+
+                for ( const auto& s : util::split(arg, ",") ) {
+                    if ( ! driver_options->logger->debugEnable(s) )
+                        return hilti::result::Error(util::fmt("Unknown debug stream '%s', use 'help' for list", arg));
+                }
+
+                break;
+            }
+
+            case 'L': compiler_options->library_paths.emplace_back(std::string(optarg)); break;
+
+            case 'o': driver_options->output_path = std::string(optarg); break;
+
+            case 'O': compiler_options->optimize = true; break;
+
+            case 'R': driver_options->report_times = true; break;
+
+            case 'T': driver_options->keep_tmps = true; break;
+
+            case 'v':
+                std::cerr << "spicyz"
+                          << " v" << hilti::configuration().version_string_long << std::endl;
+                return Nothing();
+
+            case 'h': usage(); return Nothing();
+
+            default: usage(); return hilti::result::Error(util::fmt("option %c not implemented", c));
+        }
+    }
+
+    while ( optind < argc )
+        driver_options->inputs.emplace_back(argv[optind++]);
+
+    if ( driver_options->inputs.empty() )
+        return hilti::result::Error("no input file given");
+
+    if ( ! compiler_options->debug ) {
+        if ( compiler_options->debug_trace || compiler_options->debug_flow )
+            return hilti::result::Error("cannot use --optimize with --cgdebug");
```

Comment:
Shouldn't this be `--debug-addl`?

Reply:
Turns tout the whole if() isn't needed, `debug` is set automatically with these.

Reply:
Turns out we don't need this if-clause at all, debug is always implicitly set here.